### PR TITLE
[v16] Add types to some components in the design package

### PR DESCRIPTION
**Pull request by bl-nero**
Backport #41969 to branch/v16

This comes quite late after the initial PR, because I somehow didn't backport it, and I've just hit a roadbump with backporting another change.

Needs to be merged together with https://github.com/gravitational/teleport.e/pull/4927.

### CODE REVIEW

**Comment by bl-nero** [comment]
All non-trivial changes verified using Storybook, the web app, and/or Teleterm.

**Review by ravicious — APPROVED** [review]
Did a quick test of Connect and it seems to work fine! Also damn, the buttons in unified resources on master look so much better compared to v16. ;)

**Review by avatus — APPROVED** [review]

File: web/packages/design/src/DataTable/StyledTable.tsx
[the remark this answers is not in the document]
Comment (bl-nero) [reply web/packages/design/src/DataTable/StyledTable.tsx:1]: Hmm... I don't know how exactly this happened, perhaps it was my mistake when copying from Git output.

[the remark this answers is not in the document]
Comment (bl-nero) [reply web/packages/design/src/DataTable/StyledTable.tsx:1]: OK, this list is completely bogus. I must have confused something in my notes, sorry for that. :(

**Pull request merged**